File: src/posidon/potassium/universe/Globals.java
package posidon.potassium.universe;

import posidon.potassium.backend.Sender;
import posidon.potassium.packets.Packet;

public class Globals {

    private static double time;
    private static int timeSpeed = 1;

    private static final int MAX_TIME = 24000;

    static void tick() { time = (time < MAX_TIME) ? time + timeSpeed : 0; }

    public static boolean isNight() {
        return false;
    }

    public static boolean isDay() {
        return false;
    }

    public static double getTime() { return time; }
    public static void setTime(double time) {
        Globals.time = time;
        Packet update = new Packet();
        update.put("time", Globals.time);
        Sender.sendToAllPlayers(update);
    }

    public static int getTimeSpeed() { return timeSpeed; }
    public static void setTimeSpeed(int timeSpeed) {
        Globals.timeSpeed = timeSpeed;
        Packet update = new Packet();
        update.put("timeSpeed", Globals.timeSpeed);
        Sender.sendToAllPlayers(update);
    }
}

File: src/posidon/potassium/universe/Globals.kt
package posidon.potassium.universe

import posidon.potassium.backend.Sender
import posidon.potassium.packets.Packet

object Globals {

    @JvmStatic
    var time = 0.0
        set(value) {
            field = value
            val update = Packet()
            update["time"] = time
            Sender.sendToAllPlayers(update)
        }

    var timeSpeed = 1
        set(value) {
            field = value
            val update = Packet()
            update["timeSpeed"] = timeSpeed
            Sender.sendToAllPlayers(update)
        }

    private const val MAX_TIME = 24000

    fun tick() {
        time = if (time < MAX_TIME) time + timeSpeed else 0.0
    }

    val isNight: Boolean
        get() = false

    val isDay: Boolean
        get() = false
}